server/src/Routes/Customers.hs

```diff
@@ -204,32 +204,44 @@ type RegisterRoute =
 
 registrationRoute :: RegistrationParameters -> App ()
 registrationRoute parameters = do
-    _customer <- createNewCustomer Nothing parameters
+    _customer <- createNewCustomer False parameters
     pure ()
 
-createNewCustomer :: Maybe AvalaraCustomerCode -> RegistrationParameters -> App (Entity Customer)
-createNewCustomer mbAvalaraCode = validate >=> \parameters -> do
+createNewCustomer :: Bool -> RegistrationParameters -> App (Entity Customer)
+createNewCustomer ephemeral = validate >=> \parameters -> do
     encryptedPass <- hashPassword $ rpPassword parameters
     (customer, maybeCustomerId) <- runDB $ do
-        authToken <- generateUniqueToken UniqueToken
-        let customer = Customer
-                { customerEmail = rpEmail parameters
-                , customerStoreCredit = Cents 0
-                , customerMemberNumber = ""
-                , customerEncryptedPassword = encryptedPass
-                , customerAuthToken = authToken
-                , customerStripeId = Nothing
-                , customerHelcimCustomerId = Nothing
-                , customerAvalaraCode = mbAvalaraCode
-                , customerIsAdmin = False
-                , customerVerified = False
-                }
-        (customer,) <$> insertUnique customer
+        mCustomer <- getCustomerByEmail (rpEmail parameters)
```

Comment:
The name is now kinda misleading, this function doesn't always create the new account, but rather creates or updates the existing one

Reply:
Perhaps, a name `registerNewCustomer` with a comment on how it handles "ephemeral" customers would be more suitable